react/components/ProductImages/components/Carousel/index.js

```diff
@@ -281,7 +262,7 @@ class Carousel extends Component {
             }
           )}
         >
-          <Swiper {...galleryParams}>
+          <Swiper {...galleryParams} ref={this.gallerySwiper}>
```

Comment:
Why is the `ref`s needed now?

Reply:
to remove the need for this code
```javascript
linkGallery = () => {
  const _this = this
  return function() {
    _this.setState({ gallerySwiper: this })
  }
}
```

This also removes the need to keep the `swiper` instance in the component state.